common/stringop.c

```diff
@@ -315,22 +311,25 @@ char *join_list(list_t *list, char *separator) {
 		len += (list->length - 1) * sep_len;
 	}
 
-	for (int i = 0; i < list->length; i++) {
-		len += strlen(list->items[i]);
+	for (size_t i = 0; i < list->length; i++) {
+		char *item = *(char **)list_get(list, i);
+		len += strlen(item);
 	}
 
 	char *res = malloc(len);
 
-	char *p = res + strlen(list->items[0]);
-	strcpy(res, list->items[0]);
+	char *item = *(char **)list_get(list, 0);
+	char *p = res + strlen(item);
+	strcpy(res, item);
 
-	for (int i = 1; i < list->length; i++) {
+	for (size_t i = 1; i < list->length; i++) {
+		item = *(char **)list_get(list, i);
```

Comment:
Whoops, I get I forgot to run my search-and-replace on the common code. Yeah, it's supposed to use the list_getp function.